apps/server/src/modules/authorization/authorization.service.spec.ts

```diff
@@ -32,325 +24,387 @@ describe('AuthorizationService', () => {
+
+				expect(result).toBe(true);
 			});
+		});
+	});
 
-			it('school', () => {
-				const { context, school, user } = setup();
+	describe('checkIfAuthorizedByReferences', () => {
+		describe('when isAuthorizedByReferences returns false', () => {
+			const setup = () => {
+				const context = AuthorizationContextBuilder.read([]);
+				const userId = 'test';
+				const entityId = 'test';
+				const entityName = AllowedAuthorizationEntityType.Course;
 
-				const response = service.hasPermission(user, school, context);
+				jest.spyOn(service, 'isAuthorizedByReferences').mockResolvedValueOnce(false);
 
-				expect(response).toBe(true);
+				return { context, userId, entityId, entityName };
+			};
+
+			it('should reject with ForbiddenException', async () => {
+				const { context, userId, entityId, entityName } = setup();
+
+				await expect(service.checkIfAuthorizedByReferences(userId, entityName, entityId, context)).rejects.toThrow(
+					ForbiddenException
+				);
 			});
+		});
 
-			it('user', () => {
-				const { context, user } = setup();
+		describe('when isAuthorizedByReferences returns true', () => {
+			const setup = () => {
+				const context = AuthorizationContextBuilder.read([]);
+				const userId = 'test';
+				const entityId = 'test';
+				const entityName = AllowedAuthorizationEntityType.Course;
+
+				const spy = jest.spyOn(service, 'isAuthorizedByReferences').mockResolvedValueOnce(true);
 
-				const response = service.hasPermission(user, user, context);
+				return { context, userId, entityId, entityName, spy };
+			};
 
-				expect(response).toBe(true);
+			it('should resolve', async () => {
+				const { context, userId, entityId, entityName, spy } = setup();
+
+				await expect(
+					service.checkIfAuthorizedByReferences(userId, entityName, entityId, context)
+				).resolves.not.toThrow();
+
+				spy.mockRestore();
 			});
+		});
+	});
 
-			it('team', () => {
-				const { team, user } = setup();
+	describe('isAuthorizedByReferences', () => {
+		describe('when referenceLoader throws an error', () => {
+			const setup = () => {
+				const context = AuthorizationContextBuilder.read([]);
+				const userId = 'test';
+				const entityId = 'test';
+				const entityName = AllowedAuthorizationEntityType.Course;
 
-				const context = PermissionContextBuilder.read([Permission.CHANGE_TEAM_ROLES]);
-				const response = service.hasPermission(user, team, context);
+				loader.loadEntity.mockRejectedValueOnce(InternalServerErrorException);
 
-				expect(response).toBe(true);
+				return { context, userId, entityId, entityName };
+			};
+
+			it('should reject with ForbiddenException', async () => {
+				const { context, userId, entityId, entityName } = setup();
+
+				await expect(service.isAuthorizedByReferences(userId, entityName, entityId, context)).rejects.toThrow(
+					ForbiddenException
+				);
 			});
+		});
+
+		describe('when isAuthorized of RuleManager returns true', () => {
```

Comment:
Der Nachteil daran, dass so explizit in die description zu schreiben, ist bei einem Refactoring, dass man es auch anpassen muss, wenn der Name der Methode sich ändert. Deswegen doch lieber bei etwas eher fachlichen bleiben?

Reply:
Guter Punkt! Hab die Descriptions jetzt etwas generischer gemacht. Wie ich sie "fachlich" machen soll, wusste ich gerade nicht. Kannst ja mal drüber gucken und gerne nen Vorschlag machen!